libraries/rig-package/src/RigConfig.ts

```diff
@@ -0,0 +1,270 @@
+   */
+  projectFolderPath: string;
+
+  /**
+   * A function that implements Node.js module resolution.  The {@link RigConfig.getResolvedProfileFolder}
+   * API cannot be used if this is omitted.
+   */
+  moduleResolver?: ModuleResolver;
+}
+
+/**
+ * This is the main API for loading the `config/rig.json` file format.
+ *
+ * @public
+ */
+export class RigConfig {
+  // For syntax details, see PackageNameParser from @rushstack/node-core-library
+  private static readonly _packageNameRegExp: RegExp = /^(@[A-Za-z0-9\-_\.]+\/)?[A-Za-z0-9\-_\.]+$/;
+
+  // Rig package names must have the "-rig" suffix.
+  // Also silently accept "-rig-test" for our build test projects.
+  private static readonly _rigNameRegExp: RegExp = /-rig(-test)?$/;
+
+  // Profiles must be lowercase alphanumeric words separated by hyphens
+  private static readonly _profileNameRegExp: RegExp = /^[a-z0-9_\.]+(\-[a-z0-9_\.]+)*$/;
+
+  /**
+   * Returns the absolute path of the `rig.schema.json` JSON schema file for `config/rig.json`,
+   * which is bundled with this NPM package.
+   *
+   * @remarks
+   * The `RigConfig` class already performs schema validation when loading `rig.json`; however
+   * this schema file may be useful for integration with other validation tools.
+   *
+   * @public
+   */
+  public static jsonSchemaPath: string = path.resolve(__dirname, './schemas/rig.schema.json');
+  private static _jsonSchemaObject: object | undefined = undefined;
+
+  /**
+   * The project folder path that was passed to {@link RigConfig.loadForProjectFolder}.
+   *
+   * @remarks
+   * Example: `/path/to/your-project`
+   */
+  public readonly projectFolderPath: string;
+
+  private readonly _moduleResolver: ModuleResolver | undefined;
+
+  /**
+   * Returns `true` if `config/rig.json` was found, or `false` otherwise.
+   */
+  public readonly enabled: boolean;
+
+  /**
+   * The full path to the `rig.json` file that was found, or `""` if none was found.
+   *
+   * @remarks
+   * Example: `/path/to/your-project/config/rig.json`
+   */
+  public readonly filePath: string;
+
+  /**
+   * The `"rigPackageName"` field from `rig.json`, or `""` if the file was not found.
+   *
+   * @remarks
+   * The name must be a valid NPM package name, and must end with the `-rig` suffix.
+   *
+   * Example: `example-rig`
+   */
+  public readonly rigPackageName: string;
+
+  /**
+   * The `"rigProfile"` value that was loaded from `rig.json`, or `""` if the file was not found.
+   *
+   * @remarks
+   * The name must consist of lowercase alphanumeric words separated by hyphens, for example `"sample-profile"`.
+   * If the `rig.json` file exists, but the `"rigProfile"` is not specified, then the profile
+   * name will be `"default"`.
+   *
+   * Example: `example-profile`
+   */
+  public readonly rigProfile: string;
+
+  /**
+   * The relative path to the rig profile specified by `rig.json`, or `""` if the file was not found.
```

Comment:
Why not `undefined` if the file was not found?

Reply:
Because nobody will read that property unless the file was found. So we avoid a needless `!` operator.